File: 003_largest_prime_factor.py
# What is the largest prime factor of the number 600851475143 ?

def find_largest_prime(num):
	counter = 2
	while counter < num:
		if num % counter == 0:
			num = num / counter
			counter = 1
		counter += 1
	return num

print(find_largest_prime(600851475143))

# 6857

File: 003_largest_prime_factor.php
<?php

# What is the largest prime factor of the number 600851475143 ?

$num = 600851475143;
$counter = 2;
while ($counter < $num) {
	if ($num % $counter == 0) {
		$num = $num / $counter;
		$counter = 1;
	}
	$counter += 1;
}

echo $num;
echo "\n";

# 6857